File: Algo/DynamicProgramming/TakeYouForward/practice/tuf/Sorting/Bubblesort.java
package tuf.Sorting;

import java.util.Arrays;
import java.util.stream.Collectors;

public class Bubblesort {

    int[] arr ={4,2,0,1,23,45234,-12,23343};

    public static void main(String[] args) {
        new Bubblesort();
    }


    Bubblesort(){
        sort(arr);
        print(arr);
    }

    public void sort(int[] arr){
        int N= arr.length;
        for(int i=0;i<N;i++){
            for(int j=1;j<N;j++){
                if(arr[j-1]>arr[j]) swap(arr,j-1,j);
            }
        }
    }

    public void swap(int[] arr,int src,int dest){
        int temp = arr[src];
        arr[src]= arr[dest];
        arr[dest]=temp;

}

    public void print(int[] arr){
        System.out.println(Arrays.stream(arr).boxed().collect(Collectors.toList()));
    }
}

File: Algo/DynamicProgramming/TakeYouForward/practice/practice3/Sorting/bubblesort.go
package main

import "fmt"

func swap1(src int, dest int, arr [8]int) [8]int {
	temp := arr[src]
	arr[src] = arr[dest]
	arr[dest] = temp
	return arr
}

func main1() {
	fmt.Println("Bubble Sort!!")
	arr := [8]int{3, 2, 8, 1, 6, 3, 9, 5}
	n := len(arr)
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			fmt.Printf(" i=%d, j=%d \n", i, j)
			if arr[i] > arr[j] {
				arr = swap(i, j, arr)
			}
		}
	}
	fmt.Print(arr)
}
